starter-code/index.js

```diff
@@ -1,11 +1,54 @@
 class SortedList {
-    constructor() {}
-    add(item) {}
-    get(pos) {}
-    max() {}
-    min() {}
-    average() {}
-    sum() {}
-  };
+
+    constructor() {
+    this.items = [];
+    this.length = 0;
+    }
+
+    add(item) {
+      this.items.push(item);
+      this.length += 1;
+    }
+    
+    get(pos) {
+      return this.items[pos];
+    }
+    max() {
+      if (this.length === 0) {
+        throw new Error("EmptySortedList")
+      }
+      return Math.max(...this.items);
+    }
+    
+
+    min() {
+
+    if (this.length === 0) {
+      throw new Error("EmptySortedList")
+    }
+      return Math.min(...this.items);
+    }
+    average() {
+      if (this.length === 0) {
+        throw new Error("EmptySortedList")
+      }
+
+      return this.items.reduce(function(a,b){
+        return a+b} , 0) / this.length;
+
+    }
```

Comment:
Ici ça fonctionne, mais comme plus bas tu définis déjà ta fonction sum, tu pourrais simplement l'utiliser et faire ça à la ligne 36 :
`return this.sum(this.items)/this.length;`